plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -37,11 +38,28 @@ func NewSelectorSpreadPriority(serviceLister algorithm.ServiceLister, controller
 	return selectorSpread.CalculateSpreadPriority
 }
 
+// Helper function that builds a string identifier that is unique per failure-zone
+// Returns empty-string for no zone
+func getZoneKey(node *api.Node) string {
+	labels := node.Labels
+	if labels == nil {
+		return ""
+	}
+
+	region, _ := labels[unversioned.LabelZoneRegion]
+	failureDomain, _ := labels[unversioned.LabelZoneFailureDomain]
+
+	if region == "" && failureDomain == "" {
+		return ""
+	}
+
+	return region + ":" + failureDomain
```

Comment:
In theory there could be a poor scheduling decision if there was a conflict, though I would be very surprised if this happened in the real world.  I'm changing the delimiter to ":\x00:"; I think not including a null is a very safe assumption (particularly given the consequence is just a suboptimal zone spreading).  Nicely, the null character is not printed, so the key prints as e.g. "us-east-1::us-east-1a"